lib/msf/core/post_mixin.rb

```diff
@@ -183,91 +185,37 @@ def session_incompatibility(sess_or_sid)
-      end
-
-      unmatched_wildcards = cmd_name_wildcards.select { |cmd_name_wildcard| cmd_names.none? { |cmd_name| ::File.fnmatch(cmd_name_wildcard, cmd_name) } }
-      unless unmatched_wildcards.empty?
-        # This implies that there was a typo in one of the wildcards because it didn't match anything. This is a developer mistake.
-        wlog("The #{fullname} module specified the following Meterpreter command wildcards that did not match anything: #{ unmatched_wildcards.join(', ') }")
-      end
-
-      cmd_ids = cmd_names.map { |name| Rex::Post::Meterpreter::CommandMapper.get_command_id(name) }
-
-      # XXX: Remove this condition once the payloads gem has had another major version bump from 2.x to 3.x and
-      # rapid7/metasploit-payloads#451 has been landed to correct the `enumextcmd` behavior on Windows. Until then, skip
-      # proactive validation of Windows core commands. This is not the only instance of this workaround.
-      if s.base_platform == 'windows'
-        cmd_ids = cmd_ids.select do |cmd_id|
-          !cmd_id.between?(
-            Rex::Post::Meterpreter::ClientCore.extension_id,
-            Rex::Post::Meterpreter::ClientCore.extension_id + Rex::Post::Meterpreter::COMMAND_ID_RANGE - 1
-          )
-        end
-      end
-
-      missing_cmd_ids = (cmd_ids - s.commands)
-      unless missing_cmd_ids.empty?
-        # If there are missing commands, try to load the necessary extension.
-
-        # If core_loadlib isn't supported, then extensions can't be loaded
-        return 'missing Meterpreter features: core can not be extended' unless s.commands.include?(Rex::Post::Meterpreter::COMMAND_ID_CORE_LOADLIB)
-
-        # Since core is already loaded, if the missing command is a core command then it's truly missing
-        missing_core_cmd_ids = missing_cmd_ids.select do |cmd_id|
-          cmd_id.between?(
-            Rex::Post::Meterpreter::ClientCore.extension_id,
-            Rex::Post::Meterpreter::ClientCore.extension_id + Rex::Post::Meterpreter::COMMAND_ID_RANGE - 1
-          )
-        end
-        if missing_core_cmd_ids.any?
-          return "missing Meterpreter features: #{command_names_for(missing_core_cmd_ids)}"
-        end
-
-        missing_extensions = missing_cmd_ids.map { |cmd_id| Rex::Post::Meterpreter::ExtensionMapper.get_extension_name(cmd_id) }.uniq
-        missing_extensions.each do |ext_name|
-          # If the extension is already loaded, the command is truly missing
-          return "missing Meterpreter features: #{command_names_for(missing_cmd_ids)}" if s.ext.aliases.include?(ext_name)
-
-          begin
-            s.core.use(ext_name)
-          rescue RuntimeError
-            return "unloadable Meterpreter extension: #{ext_name}"
-          end
-        end
-      end
-      missing_cmd_ids -= s.commands
-      return "missing Meterpreter features: #{command_names_for(missing_cmd_ids)}" unless missing_cmd_ids.empty?
```

Comment:
This was just extracted into a method below, and it returns an array of issues instead of a string